packages/shopify-checkout/src/client/actions/queryCheckout.ts

```diff
@@ -0,0 +1,36 @@
+import { GqlClient, ShopifyResponse } from '../../checkout-client.types';
+
+export interface CheckoutQueryParams {
+  gqlClient: GqlClient;
+  query: string;
+  input?: Record<string, unknown>;
+  queueToken?: string;
+}
+
+export default async function queryCheckout({
+  gqlClient,
+  query,
+  input,
+  queueToken
+}: CheckoutQueryParams): Promise<ShopifyResponse<unknown> | void> {
+  const variables = {
+    ...input,
+    queueToken
+  };
+
+  try {
+    const shopifyResponse = await gqlClient<
+      Pick<CheckoutQueryParams, 'input' | 'queueToken'>,
+      ShopifyResponse<unknown>
+    >({
+      query,
+      variables
+    }).catch((err) => {
+      throw new Error(err);
+    });
+
+    return shopifyResponse;
+  } catch (err) {
+    throw new Error(String(err));
+  }
+}
```

Comment:
1. Because we're not doing anything with the query response (e.g. checking it for `checkoutUserErrors` etc.), we can remove the `try/catch` and simplify this function.
2. We can simplify it further by leaving error handling entirely up to the merchant developer, which IMHO makes sense for "custom queries/mutations". This should help with the test coverage drop as well.
3. For arbitrary queries/mutations, we can't assume that `queueToken` will be a valid variable.

```suggestion
export interface CheckoutQueryParams {
  gqlClient: GqlClient;
  query: string;
  variables: Record<string, unknown>;
}

export default function queryCheckout({
  gqlClient,
  query,
  variables
}: CheckoutQueryParams): Promise<ShopifyResponse<
  Record<string, unknown>
> | void> {
  return gqlClient<typeof variables, ShopifyResponse<Record<string, unknown>>>({
    query,
    variables
  });
}
```